src/provider/states/starting.rs

```diff
@@ -56,21 +56,37 @@ impl State<PodState> for Starting {
                     "Starting command: {:?} with arguments {:?}",
                     binary, os_args
                 );
+
+                // Check if environment variables were stated for the container
                 let env_variables = if let Some(vars) = container.env() {
                     debug!(
                         "Got environment vars: {:?} service {}",
                         vars, pod_state.service_name
                     );
-                    vars.iter()
+                    match vars
+                        .iter()
                         .map(|env_var| {
-                            (
-                                String::from(&env_var.name),
-                                String::from(
-                                    &env_var.value.clone().unwrap_or_else(|| String::from("")),
-                                ),
-                            )
+                            let name = String::from(&env_var.name);
+                            let value_result = CreatingConfig::render_config_template(
+                                &template_data,
+                                &env_var.value.clone().unwrap_or_else(|| String::from("")),
+                            );
+                            match value_result {
+                                Ok(value) => Ok((name, value)),
+                                Err(error) => Err(error),
+                            }
                         })
-                        .collect::<Vec<_>>()
+                        .collect()
+                    {
```

Comment:
I find/found it to be a bit confusing to have this whole block after the match keyword.
You could pull it out into a temporary variable.
Also a comment what's happening here would be good.
I still don't quite understand what's going on to be honest.
You iterate over vars, do some mapping and then collect to...something which you match over.
What is it you're collecting into? And why can the result be a single value (`rendered_value`) below?